### PR TITLE
Implemented resource for merge request approvals

### PR DESCRIPTION
**Pull request by lorenzo**
Implemented a new resource `gitlab_merge_request_approvals` to control project options around MR approvals.

I haven't written any tests yet, but I'm already using the feature with no problems. Wanted to validate the approach with you before investing the time in writing the tests.

### CODE REVIEW

File: gitlab/resource_gitlab_merge_request_approvals.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	}
+
+	log.Printf("[DEBUG] create gitlab approvals configuration for project %s", project)
+
+	_, _, err := client.Projects.ChangeApprovalConfiguration(project, options)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(strings.Replace(project, "/", ":", 1))
```
Comment (lorenzo) [inline gitlab/resource_gitlab_merge_request_approvals.go]: This is the part I'm most unsure about. Terraform seems to enforce Ids to have a colon in them, so I had to made this up. This is obviously going to fail if the id is just a number and not the `owner/repo` path that I use in my configuration. What's a good alternative?

Comment (roidelapluie) [reply gitlab/resource_gitlab_merge_request_approvals.go]: hash the name?

**Comment by fteem** [comment]
Hi @lorenzo @roidelapluie 👋 

I find the lack of the ability to disable approvals removal on push (called `reset_approvals_on_push` in this MR) annoying, so I'd like to help out and push this further. Is this PR obsolete by now and someone else if implementing this functionality? Or maybe it's still valid and it requires more work?

**Comment by ringods** [comment]
Hello @lorenzo, I became a maintainer of this provider since a short time. I'm going over all the open pull requests one by one.

Are you still interested in getting this work in? If so, could you first rebase your work with the latest from master?

**Comment by nicholasklick** [comment]
@lorenzo closing this due to lack of response. Please resubmit if desired.

**Comment by vancouverwill** [comment]
> @lorenzo closing this due to lack of response. Please resubmit if desired.

we would be keen to get this in as find it really unproductive to have to keep going back to reapprove MR already after very minor changes

**Pull request closed without merging**